### PR TITLE
chore: add setup-bun action to workflows

### PR DESCRIPTION
**Pull request by aykutkardas**
It might be better to use the official `bun` action directly for the workflow.

In this way, we will not need to install the `setup-node` in actions where `npm` is not required. 

Just `bun` may be enough.

### CODE REVIEW

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -423,15 +436,14 @@ jobs:
...
 
-      - name: Install bun
-        run: npm install -g bun
-
       - uses: pnpm/action-setup@v2
         with:
           version: latest
 
-      - name: Install bun
-        run: npm install -g bun
```
Comment (aykutkardas) [inline .github/workflows/tests.yaml:434]: I'm not sure if this is a duplicate or necessary.

Comment (ogzhanolguncu) [reply .github/workflows/tests.yaml:434]: Probably duplicate.

**Comment by ogzhanolguncu** [comment]
Good catch. We migrated all of our SDKs in October, and somehow we missed oven-sh/setup-bun@v1, which appears to have been released in the same month.

**Comment by ogzhanolguncu** [comment]
Feel free to do this adjustment in other SDKs. PRs are welcome.

**Review by ogzhanolguncu — APPROVED** [review]

**Comment by aykutkardas** [comment]
> _"oven-sh/setup-bun@v1 is not allowed to be used in upstash/upstash-redis."_

I think you will need to update your actions policy for this repo. @ogzhanolguncu

**Review by ogzhanolguncu — APPROVED** [review]

**Review by fahreddinozcan — APPROVED** [review]

**Pull request merged**